File: app/src/main/java/com/project/personal/app_bank/API/RetrofitClient.java
package com.project.personal.app_bank.API;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {

    private static String baseUrl = "https://bank-app-test.herokuapp.com/api/";

    private static Retrofit retrofit = null;

    public static Retrofit getInstance() {

        if(retrofit==null){

            retrofit = new Retrofit.Builder().baseUrl(baseUrl).addConverterFactory(GsonConverterFactory.create()).build();
        }

        return retrofit;
    }

    public static APIInterface getListService(){
        return getInstance().create(APIInterface.class);
    }
}

File: BankApp/app/src/main/java/com/lucianogiardino/bankapp/data/RetrofitClient.kt
package com.lucianogiardino.bankapp.data

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClient {

    private const val BASE_URL = "https://bank-app-test.herokuapp.com/api"

    val instance: BankApi by lazy {
        val retrofit = Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()

        retrofit.create(BankApi::class.java)
    }

    }
